WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1734,7 +1734,14 @@ public boolean updatePostContentNewEditor(boolean isAutoSave, String title, Stri
         }
 
         boolean titleChanged = PostUtils.updatePostTitleIfDifferent(mPost, title);
-        boolean contentChanged = PostUtils.updatePostContentIfDifferent(mPost, content);
+        boolean contentChanged = true;
+        if (mEditorFragment instanceof AztecEditorFragment) {
+            contentChanged = ((AztecEditorFragment)mEditorFragment).hasHistory();
+        }
+
+        if (contentChanged) {
+            contentChanged = PostUtils.updatePostContentIfDifferent(mPost, content);
```

Comment:
I think we can skip the call to [updatePostContentIfDifferent()](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/6599-local-changes-on-web-written-posts/WordPress/src/main/java/org/wordpress/android/ui/posts/PostUtils.java#L322) since all that does is update the passed post model if the content is changed, and we already know the content is changed. Seems like we could just use `mPost.setContent(content)` if we know the content has changed and skip the potentially expensive string comparison.

Reply:
"strictly" addressed in 602a29c, but then realized that it wouldn't be future-proof anymore (i.e. if history could be disabled, then it would never save it). So added these checks in 6fc5a69. What do you think?

Reply:
Yeah, that's why I was asking about history - if it was disabled, the changes would never be saved. Adding that check probably isn't necessary but I feel safer having it there :)